R/impact_central.R

```diff
@@ -430,54 +440,55 @@ impact_by_year_of_vaccination_activity_type <- function(
     stop("Focal burden must have only one activity_type.")
   }
 
-  ## Filter FVPs to relevant year
-  fvps <- fvps[fvps$year %in% vaccination_years, ]
-  if (nrow(fvps) == 0) {
-    stop("No FVP data for this range of vaccination years.")
+  ## Filter FVPs to relevant year and aggregate
+  tot_fvps <- fvps %>%
+    dplyr::filter(.data$year %in% vaccination_years) %>%
+    dplyr::group_by(.data$country, .data$activity_type) %>%
+    dplyr::summarise(fvps = sum(.data$fvps, na.rm = TRUE))
+  if (nrow(tot_fvps) == 0) {
```

Comment:
Similar to function above, I expect checking things like this about the data will force the data to be fetched from the DB.

Reply:
It surely must cause it to be fetched, though that query would be executed on the db and only a few rows here would be transferred. You could be more efficient by doing a count here rather than returning the count rather than the rows which we don't actually need. But missed the meaning of the previous comment so may be misunderstanding here

Reply:
Yeah, same with `assert` methods above - can think about this when I come to writing that version